src/mysql/MySQL.chpl

```diff
@@ -0,0 +1,677 @@
+            mysql_query(this._cptr_mysqlconn, "START TRANSACTION;");
+        }
+
+        /*
+        Commits the current transaction.
+        This method should be called to manually commit, if autocommit is disabled,
+        and after `beginTransaction()` to commit the changes.
+        After calling `commit()`, the current transaction is committed and autocommit is
+        enabled again if it was enabled previously (that is, before the last call
+        to `beginTransaction()`).
+        */
+        override proc commit() {
+            mysql_commit(this._cptr_mysqlconn);
+
+            // If autocommit was enabled previously and is now disabled, enable it
+            // TODO: Check if it causes unintended consequences
+            if (!this.isAutocommit() && (this._prev_autocommit_state == true)) {
+                this.setAutocommit(true);
+            }
+        }
+
+        /*
+        Rolls back the current transaction.
+        After calling `rollback()`, the current transaction is committed and autocommit is
+        enabled again if it was enabled previously (that is, before the last call
+        to `beginTransaction()`).
+        */
+        override proc rollback() {
+            mysql_rollback(this._cptr_mysqlconn);
+
+            // If autocommit was enabled previously and is now disabled, enable it
+            // TODO: Check if it causes unintended consequences
+            if (!this.isAutocommit() && (this._prev_autocommit_state == true)) {
+                this.setAutocommit(true);
+            }
+        }
+    }
+
+    /*
+    Class for a MySQL result field.
+    */
+    class MySQLField : IField {
+        pragma "no doc"
+        var _fieldIdx: int(32);
+
+        pragma "no doc"
+        var _fieldName: string;
+
+        pragma "no doc"
+        var _fieldType: MySQLFieldType;
+
+        proc init(fieldIdx: int(32), fieldName: string, fieldType: MySQLFieldType) {
+            this._fieldIdx = fieldIdx;
+            this._fieldName = fieldName;
+            this._fieldType = fieldType;
+        }
+
+        override proc getFieldName(): string {
+            return this._fieldName;
+        }
+
+        override proc getFieldIdx(): int(32) {
+            return this._fieldIdx;
+        }
+
+        override proc getFieldType(): MySQLFieldType {
+            return this._fieldType;
+        }
+    }
+
+    /*
+    Class representing a row of result.
+    */
+    class MySQLRow : IRow {
+
+        // Note: the problem here is that we can't use one single function
+        // to return values of different types
+        // Because of this, even if we know the field type, we cannot really
+        // convert the value to that type and return it
+        // because that would mean returning different types from the same function
+
+        pragma "no doc"
+        var _cptr_row: MYSQL_ROW;
+        var _cptr_fields: c_ptr(MYSQL_FIELD);
+        var _fieldTypes: string;
+        var valid: bool;
```

Comment:
+nit: `_` before the variable.

Reply:
Do you need `valid`?

Reply:
No, we don't; thanks for noticing this, I'd forgotten to remove it after changing the implementation a bit. (Sorry for the late addressing, a few comments slipped by under my eyes xD).